File: 4D-gis-app/types/marker.ts
export interface MarkerData {
    /**
     * 标记的唯一标识符。
     */
    id: string;

    /**
     * 标记所在的地理纬度。
     */
    latitude: number;

    /**
     * 标记所在的地理经度。
     */
    longitude: number;

    /**
     * 标记点的标题。
     */
    title: string;

    /**
     * 可选的详细描述信息。
     */
    description?: string;

    /**
     * 可选的海拔高度（单位：米）。
     */
    altitude?: number;

    /**
     * 起始时间。
     */
    startTime?: Date;

    /**
     * 可选的结束时间。
     */
    endTime?: Date;
}

File: gis-backend/src/main/java/com/project/gis/model/entity/Marker.java
package com.project.gis.model.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.locationtech.jts.geom.Point;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "markers")
public class Marker
{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @Column(columnDefinition = "text")
    private String description;

    @Column(columnDefinition = "point", nullable = false)
    private Point location;

    private BigDecimal altitude;

    @Column(name = "start_time", nullable = false)
    private LocalDateTime startTime;

    @Column(name = "end_time")
    private LocalDateTime endTime;

    // 外键：marker_type_id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "marker_type_id", nullable = false)
    private MarkerType markerType;

    // 外键：creator_id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "creator_id", nullable = false)
    private User creator;

    // 外键：owner_id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Visibility visibility;

    public enum Visibility
    {
        personal,
        all_can_see,
        shared
    }

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
